### PR TITLE
Tfa imp

### LINKED ISSUE #1204: Implement TFA enable
Pages / steps:

* [ ] Enable
* [ ] Confirm
* [ ] Confirmed

### PR DESCRIPTION
**Pull request by na3d**
Closes #1204

### CODE REVIEW

**Comment by eatonphil** [comment]
screenshot?

**Comment by na3d** [comment]
screen shot 2017-04-11 at 9 35 49 am
screen shot 2017-04-11 at 9 35 56 am
screen shot 2017-04-11 at 9 36 22 am
screen shot 2017-04-11 at 9 37 59 am
screen shot 2017-04-11 at 9 35 41 am

**Comment by na3d** [comment]
Definitely welcome feedback on wording.

File: src/api/account.js [hunk 1]
```diff
@@ -3,3 +3,13 @@ import { thunkFetch } from './apiActionReducerGenerator';
 export function setPassword(password, expires) {
   return thunkFetch.post('/account/profile/password', { password, expires });
 }
+
+export function enableTFA() {
+  return thunkFetch.post('/account/profile/tfa-enable');
+}
+export function confirmTFA(code) {
+  return thunkFetch.post('/account/profile/tfa-enable-confirm', { 'tfa-code': code });
```
Comment (eatonphil) [inline src/api/account.js]: This is being replace with tfa_code in the deploy today btw.

File: test/profile/components/TwoFactorModal.spec.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    page.instance().setState({ tfaCode: 'theCode' });
+
+    dispatch.reset();
+    await page.instance().twoFactorConfirm();
+    expect(dispatch.callCount).to.equal(1);
+    const fn = dispatch.firstCall.args[0];
+
+    await expectRequest(fn, '/account/profile/tfa-enable-confirm', {
+      method: 'POST',
+      body: { 'tfa-code': 'theCode' },
```
Comment (eatonphil) [inline test/profile/components/TwoFactorModal.spec.js]: This is being changed in the deploy today to tfa_code.

File: src/api/account.js (hunk 1 above)
Comment (na3d) [inline src/api/account.js]: +1

**Comment by na3d** [comment]
screen shot 2017-04-12 at 10 59 48 am
screen shot 2017-04-12 at 10 59 54 am
screen shot 2017-04-12 at 11 00 13 am

File: package.json [hunk 3]
```diff
@@ -107,6 +107,8 @@
     "md5": "^2.1.0",
     "moment": "^2.17.1",
     "moment-timezone": "^0.5.11",
+    "node-sass": "^3.4.2",
```
Comment (mparke) [inline package.json]: you will need to remove these from package json, then:

`rm -r node_modules`
`npm install node-sass --save`
`npm install qrious --save`

and you should also have an updated npm-shrinkwrap.json

I will make a pull with wiki/readme section about updating

**Comment by na3d** [comment]
Thank you @mparke, we will have to figure out why my environment isn't behaving with handling that later.

**Comment by mparke** [comment]
My google authenticator fails on the QR, am I missing something?

Confirmation token form field doesn't validate empty, and submits + closes

**Comment by na3d** [comment]
My phone's camera is fine with it, though the app I normally use doesn't like it, though it also hasn't been friendly with other barcodes since it's last update.

**Comment by na3d** [comment]
screen shot 2017-04-17 at 10 15 18 am

**Pull request closed without merging**